src/message.rs

```diff
@@ -145,6 +148,25 @@ macro_rules! message_args_impl {
                 })
             }
 
+            #[cfg(all(feature="gnustep_runtime",
+                      not(any(target_arch = "arm",
+                              target_arch = "x86",
+                              target_arch = "x86_64"))))]
+            unsafe fn send<T, R>(self, obj: *mut T, sel: Sel) -> R
+                   where T: Message, R: Any {
+                    let mut receiver = obj as *mut Object;
+                    let nil: *mut Object = ::std::ptr::null_mut();
+                    let ref slot = *runtime::objc_msg_lookup_sender(&mut receiver as *mut *mut Object, sel, nil);
```

Comment:
If we can implement without slots (just using `objc_msgSend`) I think that'd cut down on complexity, and maybe later we can add a new API for caching method calls that can abstract over it in a platform-independent way, using slots on GNUstep and a different approach for Apple's runtime.